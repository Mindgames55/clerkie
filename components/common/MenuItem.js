'use client';
 
import styles from './MenuItem.module.css';
import { useSelectedLayoutSegment } from 'next/navigation';
import Link from 'next/link';

export default function MenuItem({title, href, children}) {
  const segment = useSelectedLayoutSegment() || '/';
  const isActive = href === segment;

    return (
        <li>
            <Link href={href}
                  className={isActive ? styles['item-active'] : styles.item}
            >
                    {children}
                    <span className={styles.text}>{title}</span>
            </Link>
        </li>
    )
}