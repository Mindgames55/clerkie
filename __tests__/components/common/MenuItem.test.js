import { useSelectedLayoutSegment } from 'next/navigation';
import { render, screen } from '@testing-library/react';
import MenuItem from '@/components/common/MenuItem';

jest.mock('next/navigation', () => ({
    useSelectedLayoutSegment: jest.fn(), // return undefined for base route
}))
 
it('renders correctly', () => {
    const props = {
        href: '/',
        title: 'title',
    }
    const { container } = render(<MenuItem {...props} />);
    expect(container).toMatchSnapshot();
});