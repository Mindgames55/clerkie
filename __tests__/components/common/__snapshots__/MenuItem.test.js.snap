// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <li>
    <a
      class="item-active"
      href="/"
    >
      <span
        class="text"
      >
        title
      </span>
    </a>
  </li>
</div>
`;
